Replace to-be-deprecated add_dataset in extensions tutorial

rly opened the pull request:
`GroupSpec.add_dataset` will be deprecated in HDMF 6.0. See https://github.com/hdmf-dev/hdmf/pull/1333#issuecomment-3383057453

rly committed:
Replace to-be-deprecated add_dataset

oruebel commented on docs/source/extensions_tutorial/extension_examples/labmetadata_extension.rst:
```diff
@@ -71,27 +71,35 @@ schema of your extension. See :ref:`extension-spec-api` section for details on h
-        doc="Lab-specific description of the preparation of the tissue",
-        dtype='text',
-        quantity='?'
+    labmetadata_ext = NWBGroupSpec(
```
With this change, the variable `labmetadata_ext = NWBGroupSpec(` is now defined twice. It is being created above as a basic GroupSpec and then fully redefined here. The previous logic was to create the GroupSpec first and then add to it, we should either keep that logic and uses `set_dataset` instead or update the example to merge the two bullet points to avoid possible confusion.

rly replied:
OK. I think it is fine to merge the two bullet points (see updated). It's personal preference, but this is the convention that I have been using in my extensions and all recent extensions that I have seen that use the spec API do this as well.

oruebel reviewed (changes requested):
See comment

rly committed:
Clarify LabMetaData type definition instructions

Expanded instructions for defining the LabMetaData type with additional details on Groups, Datasets, and Attributes.
